bio2zarr/plink.py

```diff
@@ -1,25 +1,46 @@
+import dataclasses
 import logging
 import pathlib
 
 import numpy as np
-import zarr
 
 from bio2zarr import constants, core, vcz
 
 logger = logging.getLogger(__name__)
 
 
+@dataclasses.dataclass
+class PlinkPaths:
+    bed_path: str
+    bim_path: str
+    fam_path: str
+
+
 class PlinkFormat(vcz.Source):
     @core.requires_optional_dependency("bed_reader", "plink")
-    def __init__(self, path):
+    def __init__(self, prefix):
         import bed_reader
 
-        self._path = pathlib.Path(path)
-        self.bed = bed_reader.open_bed(path, num_threads=1, count_A1=False)
+        # TODO we will need support multiple chromosomes here to join
+        # plinks into on big zarr. So, these will require multiple
+        # bed and bim files, but should share a .fam
+        self.prefix = str(prefix)
+        paths = PlinkPaths(
+            self.prefix + ".bed",
```

Comment:
Might be nicer to have the PlinkPaths `__init__` just take the prefix?

Reply:
Pros and cons - I tend to try and keep these dataclasses as simple as possible for reasoning about serialising and so on.